include/swift/Sema/ConstraintSystem.h

```diff
@@ -5769,6 +5769,17 @@ class ConstraintSystem {
   SWIFT_DEBUG_DUMP;
   SWIFT_DEBUG_DUMPER(dump(Expr *));
 
+  /// Return ScoreKind descriptions for printing alongside non-zero ScoreKinds
+  /// in debug output.
+  std::string getScoreKindName(ScoreKind kind) const;
+
+  /// Print Score list a with brief description of any non-zero ScoreKinds.
+  ///
+  /// \c isFoundSolution prints additional output for \c (found \c solution...)
+  /// that is not printed for \c Score or \c FixedScore in the debug output.
+  void printScoreWithKind(llvm::raw_ostream &out, const Score &score,
+                          bool isFoundSolution = true) const;
```

Comment:
I think both `getScoreKindName` and `printScoreWithKind` belong to score and we probably don't need `isFoundSolution` there too, let's just print running score and fixed score (from solution) the same way.